lib/api/repository/readonly-repository.ts

```diff
@@ -288,10 +293,14 @@ export class ReadonlyRepository<TEntity extends ModelBase, TRawData = any> imple
 		}
 		else {
 			const endpoint:string = this.entityBackendConfig.parseItemQuery ? this.entityBackendConfig.parseItemQuery(itemId, this.entity, this.paris.config, params) : `${this.getEndpointName({ where: params })}/${itemId}`;
+			let currentOptions = {};
```

Comment:
This is getting repeated in like 3 places.
Maybe extract to some function somewhere? Then you can also add a quick test to make sure it works in isolation (it's a pure function).